sftpretty/helpers.py

```diff
@@ -49,50 +49,46 @@ def hash(filename, algorithm=sha3_512(), blocksize=65536):
     return algorithm.hexdigest()
 
 
-def localtree(container, localdir, remotedir, recurse=True):
+def localtree(localbase, localdir, remotedir, recurse=True):
     '''recursively descend local directory mapping the tree to a
-    dictionary container.
-
-    :param dict container: dictionary object to save directory tree
-            {localdir:
-                 [(localdir/sub-directory,
-                   remotedir/localdir/sub-directory)],}
-        {localdir: [(content path, remotedir/content path)],}
-    :param str localdir:
-        root of local directory to descend, use '.' to start at
-        :attr:`.pwd`
-    :param str remotedir:
-        root of remote directory to append localdir too
-        path
+        list of tuples.
+
+    :param str localbase: The base of all `localdir` paths
+    :param str localdir: root of local directory to descend, use '.' to
+        start at :attr:`.pwd`
+    :param str remotedir: root of remote directory to append localdir
+        to path
     :param bool recurse: *Default: True*. To recurse or not to recurse
         that is the question
 
-    :returns: None
+    :returns: List of tuples, containing pairs of local directories and
+        remote directories.
+        [
+            (local-dir/sub-dir, remote-dir/sub-dir),
+            (lcoal-dir/sub-dir/sub-dir, remote-dir/sub-dir/sub-dir),
+            ...
+        ]
 
     :raises: Exception
 
     '''
+    tree = []
     try:
-        if localdir.startswith(':', 1) or localdir.startswith('\\'):
-            localdir = PureWindowsPath(localdir)
-        else:
-            localdir = Path(localdir).expanduser().absolute()
```

Comment:
I wasn't sure what the purpose of this was, as `Path` should have everything we need here and will use a Windows or Posix subclass based on where it is run.  I'm happy to put it back if there was a particular case it was used for.

Reply:
This is incorrect. `Path.root(), Path.parent(), etc...` all have OS specific behavior.